packages/ui/src/modals/AddMembershipModal/AddMembershipModal.tsx

```diff
@@ -14,47 +15,63 @@ interface MembershipModalProps {
   onClose: () => void
 }
 
-type ModalState = 'Create' | 'Authorize' | 'SENDING' | 'SUCCESS' | 'EXTENSION_SIGN'
+type ModalState = 'PREPARE' | 'AUTHORIZE' | 'EXTENSION_SIGN' | 'SENDING' | 'SUCCESS' | 'ERROR'
+const isError = (events: EventRecord[]) => events.find(({ event: { method } }) => method === 'ExtrinsicFailed')
 
 export const AddMembershipModal = ({ onClose }: MembershipModalProps) => {
   const { api } = useApi()
   const membershipPrice = useObservable(api?.query.members.membershipPrice(), [])
-  const [state, setState] = useState<ModalState>('Create')
+  const [step, setStep] = useState<ModalState>('PREPARE')
   const [transactionParams, setParams] = useState<Member>()
-  const [, setSubscription] = useState<Subscription | undefined>(undefined)
+  const [subscription, setSubscription] = useState<Subscription | undefined>(undefined)
+
+  useEffect(() => {
+    if (subscription) {
+      return () => subscription.unsubscribe()
+    }
+  }, [subscription])
 
   const onSubmit = (params: Member) => {
-    setState('Authorize')
+    setStep('AUTHORIZE')
     setParams(params)
   }
 
   const onSign = (transaction: Observable<ISubmittableResult>) => {
     const statusCallback = (result: ISubmittableResult) => {
```

Comment:
The `statusCallback` and `ModalState` are now duplicates of the ones in `TransferModal`. Worth extracting.